From 0f56df14a9df1f5d0b4ef2303ae643df8be0407f Mon Sep 17 00:00:00 2001
From: He Chuyue <hechuyue@wxiat.com>
Date: Tue, 22 Nov 2022 13:45:42 +0800
Subject: [PATCH] sw64: fix VM_DATA_DEFAULT_FLAGS
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 0f56df14a9df1f5d0b4ef2303ae643df8be0407f
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I645T3

--------------------------------

SW64 is designed to be have non-executable data by default. However,
VM_EXEC was set in VM_DATA_DEFAULT_FLAGS, which makes stack executable
and increases security risk. In particular, it causes some tests to
fail.

Redefine VM_DATA_DEFAULT_FLAGS with VM_DATA_FLAGS_NON_EXEC to fix this
problem.

Signed-off-by: He Chuyue <hechuyue@wxiat.com>
Reviewed-by: He Sheng <hesheng@wxiat.com>
Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/asm/page.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/sw_64/include/asm/page.h b/arch/sw_64/include/asm/page.h
index dc6a89c37231..1ba34f4f5077 100644
--- a/arch/sw_64/include/asm/page.h
+++ b/arch/sw_64/include/asm/page.h
@@ -57,8 +57,7 @@ extern unsigned long __phys_addr(unsigned long);
 #define pfn_valid(pfn)		((pfn) < max_mapnr)
 #endif /* CONFIG_FLATMEM */
 
-#define VM_DATA_DEFAULT_FLAGS		(VM_READ | VM_WRITE | VM_EXEC | \
-					 VM_MAYREAD | VM_MAYWRITE | VM_MAYEXEC)
+#define VM_DATA_DEFAULT_FLAGS	VM_DATA_FLAGS_NON_EXEC
 #include <asm-generic/memory_model.h>
 #include <asm-generic/getorder.h>
 #endif
-- 
2.33.0

